ibm/service/power/resource_ibm_pi_network_security_group_rule.go

```diff
@@ -0,0 +1,611 @@
+// Copyright IBM Corp. 2024 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package power
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/power/models"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/validate"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func ResourceIBMPINetworkSecurityGroupRule() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIBMPINetworkSecurityGroupRuleCreate,
+		ReadContext:   resourceIBMPINetworkSecurityGroupRuleRead,
+		DeleteContext: resourceIBMPINetworkSecurityGroupRuleDelete,
+		Importer:      &schema.ResourceImporter{},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			// Arguments
+			Arg_Action: {
+				ConflictsWith: []string{Arg_NetworkSecurityGroupRuleID},
+				Description:   "The action to take if the rule matches network traffic.",
+				Optional:      true,
+				Type:          schema.TypeString,
+				ValidateFunc:  validate.ValidateAllowedStringValues([]string{"allow", "deny"}),
+			},
+			Arg_CloudInstanceID: {
+				Description:  "The GUID of the service instance associated with an account.",
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			Arg_DestinationPorts: {
+				Computed:      true,
+				ConflictsWith: []string{Arg_NetworkSecurityGroupRuleID},
+				Description:   "Destination port ranges.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						Arg_Maximum: {
+							Computed:    true,
+							Description: "The end of the port range, if applicable. If values are not present then all ports are in the range.",
+							Type:        schema.TypeFloat,
+						},
+						Arg_Minimum: {
+							Computed:    true,
+							Description: "The start of the port range, if applicable. If values are not present then all ports are in the range.",
+							Type:        schema.TypeFloat,
+						},
+					},
+				},
+				Optional: true,
+				Type:     schema.TypeList,
+			},
```

Comment:
I think you can keep the top level as `Arg_...` and the inner one as `Attr_...`

```suggestion
			Arg_DestinationPorts: {
				Computed:      true,
				ConflictsWith: []string{Arg_NetworkSecurityGroupRuleID},
				Description:   "Destination port ranges.",
				Elem: &schema.Resource{
					Schema: map[string]*schema.Schema{
						Att_Maximum: {
							Computed:    true,
							Description: "The end of the port range, if applicable. If values are not present then all ports are in the range.",
							Type:        schema.TypeFloat,
						},
						Att_Minimum: {
							Computed:    true,
							Description: "The start of the port range, if applicable. If values are not present then all ports are in the range.",
							Type:        schema.TypeFloat,
						},
					},
				},
				Optional: true,
				Type:     schema.TypeList,
			},
```

Reply:
But doesn't the user, provide those values, shouldn't there be a Arg and Attr version?